src/libtiled/tilelayer.h

```diff
@@ -568,6 +571,19 @@ inline void TileLayer::setSize(const QSize &size)
     mHeight = size.height();
 }
 
+inline QRect TileLayer::bounds() const
+{
+    QRect bound = mBounds.boundingRect();
+
+    int x = std::min(0, bound.x());
+    int y = std::min(0, bound.y());
+    int width = std::max(mWidth, bound.width());
+    int height = std::max(mHeight, bound.height());
+
+    return QRect(x, y, width, height).translated(mX, mY);
```

Comment:
Since in your only usage of `bounds` so far, you're translating the rectangle back to local coordinates, I wonder if it may be better to just return local coordinates instead?